libs/utils/decorate.js

```diff
@@ -0,0 +1,138 @@
+//
+// Shared decorate utils for Consonant blocks
+// Block usage ref: [media, z-pattern]
+//
+
+import { decorateLinkAnalytics } from './analytics.js';
+
+export function decorateButtons(el, isLarge) {
+  const buttons = el.querySelectorAll('em a, strong a');
+  buttons.forEach((button) => {
+    const parent = button.parentElement;
+    const buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
+    const buttonSize = isLarge ? 'button-XL' : 'button-M';
+    button.classList.add('con-button', buttonType, buttonSize);
+    parent.insertAdjacentElement('afterend', button);
+    parent.remove();
+  });
+  if (buttons.length > 0) {
+    const actionArea = buttons[0].closest('p');
+    actionArea.classList.add('action-area');
+    actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
+  }
+}
+
+export function initIcons(el) {
+  const regex = /[^{{]+(?=}})/g; // {{value}}
```

Comment:
I discussed with @auniverseaway .  Let's have all icon names start with `icon-`, and here do a search for `{{icon-*}}`

This way the icon-library.json keys don't have to start with `icon-`.

Reply:
All tokens are now stored in tokens.xls with a separate sheet for each kind of token (color, icon etc).

This will need to be refactored to be more generic in a followup PR